### DISCUSSION TITLE
10-year expiration on CA certificates

### DISCUSSION THREAD

**Discussion by splashcd** [k3s-io/k3s#6871] [Q&A] [answered]
I'm trying to determine what a good rotation frequency for CA certs would be.  They come with a 10-year shelf life.

1. Do you have thoughts on the default 10-year expiration?  Is it too long or just right?
2. Do you have thoughts on the risk that 10-year CA certs might pose to k3s installs?
3. Is there an existing automated method to rotate CA certs on K3s installs?

**Comment by brandond** [comment, marked as answer]
We do not currently have a way to alter the 10-year validity of the CA certificates, nor do we support rotating them. We are tracking improvements to this in https://github.com/k3s-io/k3s/pull/6615
